kedro/framework/cli/starters.py

```diff
@@ -95,7 +96,45 @@ class KedroStarterSpec:
 KEDRO_PATH = Path(kedro.__file__).parent
 TEMPLATE_PATH = KEDRO_PATH / "templates" / "project"
 
-_STARTERS_REPO = "git+https://github.com/kedro-org/kedro-starters.git"
+
+def _get_latest_starters_version() -> str:
+    if "STARTERS_VERSION" not in os.environ:
+        GITHUB_TOKEN = os.getenv("GITHUB_TOKEN")
+        headers = {}
+        if GITHUB_TOKEN:
+            headers["Authorization"] = f"token {GITHUB_TOKEN}"
+
+        try:
+            response = requests.get(
+                "https://api.github.com/repos/kedro-org/kedro-starters/releases/latest",
+                headers=headers,
+                timeout=10,
+            )
+            response.raise_for_status()  # Raise an HTTPError for bad status codes
+            latest_release = response.json()
+        except requests.exceptions.RequestException as e:
+            raise RuntimeError(
+                f"Error fetching kedro-starters latest release version: {e}"
+            )
```

Comment:
This piece of code is now also called in a regular `kedro new` flow outside of the whole release process, so instead of raising an exception here, I think it would be better to only log the error and just return an empty string. Otherwise users wouldn't be able to do `kedro new` when something is wrong with the GH API.

Reply:
That's a good point